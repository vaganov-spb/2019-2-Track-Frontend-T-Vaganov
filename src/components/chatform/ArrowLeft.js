import React from 'react';
import headStyles from '../../styles/ChatHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function ArrowLeft(props) {
	return (
		<div className={headStyles.arrow}>
			<span className={headStyles.left_icon}>
				{props.children}
			</span>
		</div>
	);
}

