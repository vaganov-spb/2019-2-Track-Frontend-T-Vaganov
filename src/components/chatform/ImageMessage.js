/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import ruleStyles from '../../styles/MessageRule.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function ImageMessage(props){
	const img = props.img;
	const date = props.Date;
	return (
		<div className={ruleStyles.imagewrap}>
			<img className={ruleStyles.image} src={img} alt=''/>
			<div className={ruleStyles.date}>{date}</div>
		</div>
	);
}