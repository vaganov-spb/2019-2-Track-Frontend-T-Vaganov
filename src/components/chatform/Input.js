import React from 'react';
import { connect } from 'react-redux';
import { setTime } from '../../utils';
import { getMessagesSuccess, changeMessageInputValue, clearMessageInputValue } from '../../actions';
import formStyles from '../../styles/FormInput.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function Input (props){
	function onKeyPress(event) {
		if (event.key === 'Enter') {
			event.preventDefault();
			if(props.canSend) {
				if(props.chatId !== 'Group Chat'){
					props.addMessages(
						[{message: `${props.currentText}`, type: 'text', time: `${setTime()}`}],
						props.chatId
					);
				}  else {
					fetch('http://localhost:8000/chats/1/newmessage/',{
						method: 'POST',
						headers: {'Content-Type': 'application/json'},
						body: JSON.stringify({content: `${props.currentText}`, chatId:21})
					});
				}
				props.clearInput(props.chatId);
			}
		}
	}

	function handleChange(event) {
		if(event.target.value) {
			props.changeInput(event.target.value, props.chatId);
		} else {
			props.clearInput(props.chatId);
		}
	};

	return (
		<input
			className={formStyles.form}
			type="text"
			value={props.currentText}
			placeholder="Cooбщение"
			onChange={handleChange}
			onKeyPress={onKeyPress}
		/>
	);
}


const mapStateToProps = (state, ownProps) => {
	return {
		currentText: state.currentInputText[ownProps.chatId].text,
		canSend: state.currentInputText[ownProps.chatId].flag,
	};
};

const mapDispatchToProps = (dispatch) => ({
	addMessages: (message, chatId) => dispatch(getMessagesSuccess(message, chatId)),
	changeInput: (text, chatId) => dispatch(changeMessageInputValue(text, chatId)),
	clearInput: (chatId) => dispatch(clearMessageInputValue(chatId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Input);