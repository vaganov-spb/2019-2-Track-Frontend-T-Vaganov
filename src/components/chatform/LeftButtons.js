/* eslint-disable react-hooks/exhaustive-deps */
import React from 'react';
import messageStyles from '../../styles/MessageForm.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function LeftButtons(props){
	const myRef = React.createRef();

	function fileUpload(event) {
		const uploadfile = myRef.current;
		uploadfile.click();
		event.preventDefault();
	}

	function fileTransfer(event){
		props.fileChange(event.target.files);
	}


	function onGeo() {
		// TODO: move to utils
		const options = {
			enableHighAccuracy: true,
			timeout: 5000,
			maximumAge: 0
		};
			
		const success = (pos) => {
			const crd = pos.coords;
			// TODO: get rid of this "props" dependency
			props.change(`https://www.openstreetmap.org/#map=18/${crd.latitude}/${crd.longitude}`);
			props.send();
		};
		
		const error = (err) => {
			alert('Не удалось отправить геопозицию');
		};

		navigator.geolocation.getCurrentPosition(success, error, options);
	}

	return(
		<React.Fragment>
			<input 
				type="file" 
				ref={myRef} 
				accept="image/*" 
				capture style={{display: 'None',}} 
				multiple onChange={fileTransfer}
			/>
			<img
				className={messageStyles.image}
				src="https://cdn1.iconfinder.com/data/icons/social-17/48/photos2-512.png"
				alt=""
				onClick={fileUpload}
			/>
			<img
				className={messageStyles.image}
				src="https://i.pinimg.com/originals/9c/91/98/9c919823b4cac48bec5af1f236a39efd.png"
				alt=""
				onClick={onGeo}
			/>
		</React.Fragment>
	);
}
