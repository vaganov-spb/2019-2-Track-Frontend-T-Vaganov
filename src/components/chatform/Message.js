/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import ruleStyles from '../../styles/MessageRule.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function Message(props) {
	const value = props.text;
	let content = props.text;
	if(value && value.startsWith('http')) {
		content = <a href={value}> {value}</a>;
	} 
	return (
		<div className={ruleStyles.message}>
			<div className={[ruleStyles.textMs, ruleStyles.text].join(' ')}>
				{content}
			</div>
			<div className={[ruleStyles.textMs, ruleStyles.rightText].join(' ')}>{props.Date} from {props.user}</div>
		</div>
	);
}
