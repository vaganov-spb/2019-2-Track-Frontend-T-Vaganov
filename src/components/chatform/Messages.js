/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import ruleStyles from '../../styles/MessageRule.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function Message(props) {
	const value = props.text;
	let content = props.text;
	if(value && value.startsWith('http')) {
		content = <a href={value}> {value}</a>;
	} 
	return (
		<div className={ruleStyles.message}>
			<div className={[ruleStyles.textMs, ruleStyles.text].join(' ')}>
				{content}
			</div>
			<div className={[ruleStyles.textMs, ruleStyles.rightText].join(' ')}>{props.Date} from {props.user}</div>
		</div>
	);
}

export function ImageMessage(props){
	const img = props.img;
	const date = props.Date;
	return (
		<div className={ruleStyles.imagewrap}>
			<img className={ruleStyles.image} src={img} alt=''/>
			<div className={ruleStyles.date}>{date}</div>
		</div>
	);
}

export function VoiceMessage(props) {
	return(
		<audio className={ruleStyles.audio} controls="controls">
			<source src={props.audio} />
		</audio> 
	);
}