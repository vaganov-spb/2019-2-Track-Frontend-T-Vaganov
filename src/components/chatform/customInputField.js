import React from 'react';
import { connect } from 'react-redux';
import Chat from './MessageForm';
import { getMessagesSuccess, changeMessageInputValue, clearMessageInputValue } from '../../actions';
import formStyles from '../../styles/FormInput.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function Input (props){
	function onKeyPress(event) {
		if (event.key === 'Enter' && props.canSend) {
			props.addMessages([{message: `${props.currentText}`, type: 'text', time: `${Chat.setTime()}`}], props.chatId);
			props.clearInput(props.chatId);
			event.preventDefault();
		} else if (event.key === 'Enter') {
			event.preventDefault();
		}
	}

	function handleChange(event) {
		if(event.target.value) {
			props.changeInput(event.target.value, props.chatId);
		} else {
			props.clearInput(props.chatId);
		}
	};

	return (
		<input
			className={formStyles.form}
			type="text"
			value={props.currentText}
			placeholder="Cooбщение"
			onChange={handleChange}
			onKeyPress={onKeyPress}
		/>
	);
}


const mapStateToProps = (state, ownProps) => {
	return {
		currentText: state.currentInputText[ownProps.chatId].text,
		canSend: state.currentInputText[ownProps.chatId].flag,
	};
};

const mapDispatchToProps = (dispatch) => ({
	addMessages: (message, chatId) => dispatch(getMessagesSuccess(message, chatId)),
	changeInput: (text, chatId) => dispatch(changeMessageInputValue(text, chatId)),
	clearInput: (chatId) => dispatch(clearMessageInputValue(chatId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Input);