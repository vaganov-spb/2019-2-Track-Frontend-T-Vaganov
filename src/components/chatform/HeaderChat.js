import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { HeaderTop } from '../chatslist/HeaderTop';
import { ArrowLeft } from './ArrowLeft';
import headStyles from '../../styles/ChatHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function HeaderChat(props) {
	return (
		<div className={headStyles.main_head}>
			<HeaderTop />
			<div className={headStyles.message_header_bottom}>
				<ArrowLeft>
					<Link style={{color: 'var(--white)',}} to="/">
						<i className="fa fa-arrow-left"/>
					</Link>
				</ArrowLeft>
				<div className={headStyles.message_header_img}>
					<img className={headStyles.avatar} alt="" src={props.src} />
				</div>

				<div className={headStyles.username}>
					<span className={headStyles.user}>{props.name}</span>
					<p className={headStyles.online}>в сети 2 часа назад</p>
				</div>
				<div className={headStyles.search_icons}>
					<span className={headStyles.right_icons}>
						<i className="fa fa-search" />
					</span>
					<span className={headStyles.right_icons}>
						<i className="fa fa-ellipsis-v" />
					</span>
				</div>
			</div>
		</div>
	);
}


const mapStateToProps = (state, ownProps) => {
	const chatId = ownProps.chatId;
	const obj = {
		chatId,
		name: state.chats.chats[chatId].name,
		src: state.chats.chats[chatId].url,
		isLoaded: state.chats.isLoaded,
		error: state.chats.error,
	};

	return obj;
};

export default connect(mapStateToProps)(HeaderChat);
