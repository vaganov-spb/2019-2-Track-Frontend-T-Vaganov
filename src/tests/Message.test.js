/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';  
import renderer from 'react-test-renderer'; 
import { Message } from '../components/chatform/Message';  


describe('Message renders and act correctly', () => {
	let props;
	let component;

	beforeEach(() => {
		props = {
			text: 'test_text',
			Date: 'test_Date',
			user: 'test_user',
		};
        
		component = renderer
			.create(
				<Message { ...props } />
			)
			.toJSON();
         
	});

	it ('should render normally', () => {
		expect(component).toMatchSnapshot();
	});
});