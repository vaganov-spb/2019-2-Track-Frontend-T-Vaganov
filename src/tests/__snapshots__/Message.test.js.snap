// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders and act correctly should render normally 1`] = `
<div
  className="message"
>
  <div
    className="textMs text"
  >
    test_text
  </div>
  <div
    className="textMs rightText"
  >
    test_Date
     from 
    test_user
  </div>
</div>
`;
